File: src/app/models/JobStatus.scala
package app.models

sealed trait JobStatus

object JobStatus {
  case object DRAFT extends JobStatus
  case object ACTIVE extends JobStatus
  case object COMPLETED extends JobStatus
  case object CANCELLED extends JobStatus
  case object DELETED extends JobStatus

  def apply(s: String) = s match {
    case "DRAFT" => DRAFT
    case "ACTIVE" => ACTIVE
    case "COMPLETED" => COMPLETED
    case "CANCELLED" => CANCELLED
    case "DELETED" => DELETED
  }
}

File: ui/.backup/lib/filters/jobStatus.js
export default () =>
  function jobStatus(job) {
    if (job && job.status) {
      switch (job.status) {
        case 'INACTIVE': return 'Inactive'
        case 'ACTIVE': return 'Active'
        case 'COMPLETED': return 'Completed'
        case 'CANCELLED': return 'Cancelled'
        case 'DELETED': return 'Deleted'
        default: return undefined
      }
    } else {
      return undefined
    }
  }
